## Gegevens

Beschikbare grondstoffen:

 * 300 ml essence A
 * 350 ml essence B
 * 150 ml essence C
 
Samenstellingen:

 * Moonlight: 4 ml essence A en 2 ml essence B.
 * Daydream: 4 ml essence B en 2 ml essence C.
 * Sunflower: 3 ml essence A, 2 ml essence B en 1 ml essence C.
 
Winst:

 * Moonlight: €45,-
 * Daydream: €50,-
 * Sunflower: €60,-

## Vragen

_Tableau oplossing met de simplex methode._

15) Hoeveel flesjes van welke parfum moet de componist maken om een zo groot mogelijk winst te maken?

_Gevoeligheidsanalyse toepassen op het eindtableau._

41a) Stel, de parfumeur heeft geen 150 ml, maar 160 ml van essence C tot zijn beschikking. Wat zou dan de optimale oplossing en maximale winst zijn? En hoeveel ml van essence B blijft er dan over?

41b) Wat is de optimale oplossing en maximale winst als de parfumeur 200 ml van essence C tot zijn beschikking heeft?

41c) Stel, de parfumeur heeft geen 300 ml maar slechts 240 ml van essence A tot zijn beschikking. Wat zou dan de optimale oplossing en maximale winst zijn? En hoeveel ml van essence B blijft er dan over?

41d) Als de parfumeur toch minimaal 15 flesjes Moonlight wil maken, wat zou dan de optimale oplossing en maximale winst zijn? En hoeveel ml van essence B blijft er dan over?

# Opgave 15

De beslissingsvariabelen zijn:

 * $m$: aantal flesjes Moonlight
 * $d$: aantal flesjes Daydream
 * $s$: aantal flesjes Sunflower

LP-model:

 * $\max 45m + 50d + 60s $
 * $4m + 3s \leq 300$
 * $2m + 4d + 2s \leq 350$
 * $2d+s\leq150$

De doelfunctie $D=45m+50d+60s \iff D-45m-50d-60s=0$.

Canonieke vorm:

 * $\tag{1} D-45m-50d-60s = 0 $
 * $\tag{2} 4m + 3s + s_1 = 300$
 * $\tag{3} 2m + 4d + 2s + s_2 = 350$
 * $\tag{4} 2d+s + s_3=150$

In [7]:
d = c(-10,-5,-8)
x = c(10,2,0)
y = c(0,2,3)
s1 = c(0,1,0)
s2 = c(0,0,1)
RHS = c(1,2,3)
M = cbind(d,x,y,s1,s2,RHS)
rownames(M) = c('d', 's1', 's2')
M

d   x  y s1 s2 RHS
d  -10 10 0 0  0  1  
s1  -5  2 2 1  0  2  
s2  -8  0 3 0  1  3

In [8]:
M[2,] = M[2,] + 1/2*M[3,]
rownames(M) = c('d', 'x', 's2')
M

d   x  y   s1 s2  RHS
d  -10 10 0.0 0  0.0 1.0
x   -9  2 3.5 1  0.5 3.5
s2  -8  0 3.0 0  1.0 3.0

In [9]:
D = c(1,0,0,0)
m = c(-45, 4,2,0)
d = c(-50,0,4,2)
s = c(-60,3,2,1)
s1 = c(0,1,0,0)
s2 = c(0,0,1,0)
s3 = c(0,0,0,1)
RHS = c(0,300,350,150)
M = cbind(D,m,d,s,s1,s2,s3,RHS)
rownames(M) = c('D', 's1', 's2', 's3')
print('Iteration 1')
M

[1] "Iteration 1"


D m   d   s   s1 s2 s3 RHS
D  1 -45 -50 -60 0  0  0    0
s1 0   4   0   3 1  0  0  300
s2 0   2   4   2 0  1  0  350
s3 0   0   2   1 0  0  1  150

In [10]:
M[2,] = M[2,] / 3
M[1,] = M[1,] + 60*M[2,]
M[3,] = M[3,] - 2*M[2,]
M[4,] = M[4,] - M[2,]
rownames(M) = c('D', 's', 's2', 's3')
print('Iteration 2')
M

[1] "Iteration 2"


D m          d   s s1         s2 s3 RHS 
D  1 35.0000000 -50 0 20.0000000 0  0  6000
s  0  1.3333333   0 1  0.3333333 0  0   100
s2 0 -0.6666667   4 0 -0.6666667 1  0   150
s3 0 -1.3333333   2 0 -0.3333333 0  1    50

In [11]:
M[1,] = M[1,] + 25*M[4,]
M[3,] = M[3,] - 2*M[4,]
rownames(M) = c('D', 's', 's2', 'd')
print('Iteration 3')
M

[1] "Iteration 3"


D m         d s s1         s2 s3 RHS 
D  1  1.666667 0 0 11.6666667 0  25 7250
s  0  1.333333 0 1  0.3333333 0   0  100
s2 0  2.000000 0 0  0.0000000 1  -2   50
d  0 -1.333333 2 0 -0.3333333 0   1   50

# Opgave 41

_41a) Stel, de parfumeur heeft geen 150 ml, maar 160 ml van essence C tot zijn beschikking. Wat zou dan de optimale oplossing en maximale winst zijn? En hoeveel ml van essence B blijft er dan over?_

Hiervoor moeten we vergelijking $(4)$ aanpassen.

$$\tag{4} 2d+s+s_3=150 $$

Hier voegen we een variabel $k$ aan toe, dit geeft: $2d+s+s_3=150+k$. Omschrijven geeft: $2d+s+(s_3-k)=150$. Vervolgens stellen we $s_3^*=s_3-k$ en vervangen we dit in de vergelijking:

$$ \tag{4*} 2d+s+s_3^*=150$$

Als we opnieuw het simplex tableau oplossen met deze nieuwe vergelijking, dan krijgen we hetzelfde resultaat. Het is dus niet nodig op opnieuw het tableau op te lossen. 

In [12]:
M

D m         d s s1         s2 s3 RHS 
D  1  1.666667 0 0 11.6666667 0  25 7250
s  0  1.333333 0 1  0.3333333 0   0  100
s2 0  2.000000 0 0  0.0000000 1  -2   50
d  0 -1.333333 2 0 -0.3333333 0   1   50

Vervolgens vervangen we $s_3^*$ met $s_3-k$, en schuiven we de termen naar de rechterkant. Dit geeft de volgende $\text{RHS}$: 

$$\begin{bmatrix}D\\s\\s_2\\d\end{bmatrix}=\begin{bmatrix}7250+25k\\100\\50-2k\\50+k\end{bmatrix}$$

We kunnen zien dat _essence C_ nu $50+k$ is, hier moet $10$ bij, dus $k=10$. Invullen geeft:

$$\begin{bmatrix}D\\s\\s_2\\d\end{bmatrix} = \begin{bmatrix} 7250 + 250 \\ 100 \\ 50 - 20 \\ 50 + 10 \end{bmatrix} $$

De nieuwe optimale oplossing is $(m,d,s)=(0,60,100)$ met een winst van $€7500$.